go/cmd/dolt/commands/status.go

```diff
@@ -63,215 +66,287 @@ func (cmd StatusCmd) ArgParser() *argparser.ArgParser {
+	// configure SQL engine
+	queryist, sqlCtx, closeFunc, err := cliCtx.QueryEngine(ctx)
 	if err != nil {
 		return handleStatusVErr(err)
 	}
-
-	staged, notStaged, err := diff.GetStagedUnstagedTableDeltas(ctx, roots)
-	if err != nil {
-		return handleStatusVErr(err)
+	if closeFunc != nil {
+		defer closeFunc()
 	}
 
-	ws, err := dEnv.WorkingSet(ctx)
+	// get ignored tables
+	ignorePatterns, err := getIgnoredTablePatternsFromSql(queryist, sqlCtx)
 	if err != nil {
-		handleStatusVErr(err)
+		return handleStatusVErr(err)
 	}
 
-	as, err := merge.GetMergeArtifactStatus(ctx, ws)
+	// get current branch name
+	branchName, err := getBranchName(queryist, sqlCtx)
 	if err != nil {
-		handleStatusVErr(err)
+		return handleStatusVErr(err)
 	}
 
-	err = PrintStatus(ctx, dEnv, staged, notStaged, apr.Contains(cli.ShowIgnoredFlag), as)
+	// get conflicts
+	conflicts, err := getRowsForSql(queryist, sqlCtx, "select * from dolt_conflicts;")
 	if err != nil {
 		return handleStatusVErr(err)
 	}
-	return 0
-}
-
-func PrintStatus(ctx context.Context, dEnv *env.DoltEnv, stagedTbls, notStagedTbls []diff.TableDelta, showIgnoredTables bool, as merge.ArtifactStatus) error {
-	headRef, err := dEnv.RepoStateReader().CWBHeadRef()
-	if err != nil {
-		return err
+	conflictedTables := make(map[string]bool)
+	for _, row := range conflicts {
+		tableName := row[0].(string)
+		conflictedTables[tableName] = true
 	}
 
-	cli.Printf(branchHeader, headRef.GetPath())
-
-	err = printRemoteRefTrackingInfo(ctx, dEnv)
+	// get statuses
+	statusRows, err := getRowsForSql(queryist, sqlCtx, "select * from dolt_status;")
 	if err != nil {
-		return err
+		return handleStatusVErr(err)
 	}
+	statusPresent := len(statusRows) > 0
 
-	mergeActive, err := isMergeActive(ctx, dEnv)
+	// staged/working tables
+	stagedTableNames := make(map[string]bool)
+	workingTableNames := make(map[string]bool)
+	diffs, err := getRowsForSql(queryist, sqlCtx, "select * from dolt_diff where commit_hash='WORKING' OR commit_hash='STAGED';")
 	if err != nil {
-		return err
+		return handleStatusVErr(err)
 	}
-
-	if mergeActive {
-		if as.HasConflicts() && as.HasConstraintViolations() {
-			cli.Println(fmt.Sprintf(unmergedTablesHeader, "conflicts and constraint violations"))
-		} else if as.HasConflicts() {
-			cli.Println(fmt.Sprintf(unmergedTablesHeader, "conflicts"))
-		} else if as.HasConstraintViolations() {
-			cli.Println(fmt.Sprintf(unmergedTablesHeader, "constraint violations"))
+	for _, row := range diffs {
+		commitHash := row[0].(string)
+		tableName := row[1].(string)
+		if commitHash == "STAGED" {
+			stagedTableNames[tableName] = true
 		} else {
-			cli.Println(allMergedHeader)
+			workingTableNames[tableName] = true
 		}
 	}
 
-	n := printStagedDiffs(cli.CliOut, stagedTbls, true)
-	n, err = PrintDiffsNotStaged(ctx, dEnv, cli.CliOut, notStagedTbls, true, showIgnoredTables, n, as)
+	// get merge status
+	mergeRows, err := getRowsForSql(queryist, sqlCtx, "select * from dolt_merge_status;")
```

Comment:
There are a lot of separate calls to the server for this. Ideas on how we can reduce the number of round trips?

Reply:
Removed one call that was unnecessary.
From offline convo, it was decided that there can be improvements in this area, but those can wait until after we ship these features.